packages/react-theme-provider/src/components/ThemeProvider/useThemeProvider.ts

```diff
@@ -34,9 +33,10 @@ export const useThemeProvider = (
   );
 
   const parentTheme = useTheme();
-  const localTheme = state.theme;
+  const localTheme = draftState.theme;
+
+  const state = assign(draftState, { theme: mergeThemes(parentTheme, localTheme) });
```

Comment:
Can you please clarify why this change is required? Is it only for creating `state` variables?

```suggestion
  const state = draftState;
  state.theme = mergeThemes(parentTheme, localTheme);
```

Do I correctly understand that this code does the same?